xarray/tests/test_backends.py

```diff
@@ -1539,6 +1539,15 @@ def test_roundtrip_consolidated(self):
             self.check_dtypes_roundtripped(expected, actual)
             assert_identical(expected, actual)
 
+    def test_with_chunkstore(self):
+        save_kwargs = {}
+        expected = create_test_data()
+        with self.create_zarr_target() as store_target:
+            self.save(expected, store_target, **save_kwargs)
+            open_kwargs = {'chunk_store': store_target}
+            with self.open(store_target, **open_kwargs) as ds:
```

Comment:
Here it looks like the same underlying store is being used for both the main store and the chunk store. My concern here is that this does not actually test the functionality of the `chunk_store` keyword. Imagine if `chunk_store` were just a dummy argument that did nothing? The test would pass anyway, since all the required keys would be find in `store_target`.

This test also drives home the need for a `chunk_store` argument in `to_zarr` as well.

Reply:
I added (9060925) `chunk_store` argument to the `to_zarr()` method so the test now uses separate Zarr stores.